src/Fable.ReactGoogleMaps/Fable.Helpers.ReactGoogleMaps.fs

```diff
@@ -0,0 +1,172 @@
+    | OnPlacesChanged of (unit -> unit)
+        interface ISearchBoxProperties
+
+    type IMarkerProperties =
+        interface end
+
+    [<RequireQualifiedAccess>]
+    type MarkerProperties =
+    | Key of obj
+    | Title of string
+    | Icon of string
+    | OnClick of (unit -> unit)
+    | Position of Coordinates.Position
+        interface IMarkerProperties
+
+    type IMarkerClustererProperties =
+        interface end
+
+    [<RequireQualifiedAccess>]
+    type MarkerClustererProperties =
+    | Key of obj
+    | AverageCenter of bool
+    | MaxZoom of int
+    | EnableRetinaIcons of bool
+    | GridSize of float
+        interface IMarkerClustererProperties
+
+    type IMapProperties =
+        interface end
+
+    [<RequireQualifiedAccess>]
+    type MapProperties =
+    | SetRef of (obj -> unit)
+    | ApiKey of string
+    | DefaultZoom of int
+    | SearchBoxText of string
+    | ShowSearchBox of bool
+    | ShowTrafficLayer of bool
+    | DefaultCenter of Coordinates.Position
+    | Center of Coordinates.Position
+    | OnCenterChanged of (unit -> unit)
+    | OnPlacesChanged of (Places.Place [] -> unit)
+    | OnZoomChanged of (unit -> unit)
+    | OnIdle of (unit -> unit)
+    | Markers of React.ReactElement
+    | MapLoadingContainer of string
+    | MapContainer of string
+        interface IMapProperties
+let InfoWindow: RCom = import "InfoWindow" "react-google-maps"
+
+/// A wrapper around google.maps.InfoWindow
+let infoWindow (props:Props.IInfoWindowProperties list) child : React.ReactElement =
+    R.from InfoWindow (keyValueList CaseRules.LowerFirst props) [child]
+
+let SearchBox: RCom = import "SearchBox" "react-google-maps"
+
+/// A wrapper around google.maps.places.SearchBox on the map
+let searchBox (props:Props.ISearchBoxProperties list) children : React.ReactElement =
+    R.from SearchBox (keyValueList CaseRules.LowerFirst props) children
+
+
+let Marker: RCom = import "Marker" "react-google-maps"
+
+/// A wrapper around google.maps.Marker
+let marker (props:Props.IMarkerProperties list) children : React.ReactElement =
+    R.from Marker (keyValueList CaseRules.LowerFirst props) children
+
+let MarkerClusterer: RCom = import "MarkerClusterer" "react-google-maps/lib/components/addons/MarkerClusterer.js"
+
+/// A wrapper around MarkerClusterer - https://github.com/mahnunchik/markerclustererplus/blob/master/docs/reference.html
+let markerClusterer (props:Props.IMarkerClustererProperties list) (markers:React.ReactElement list) : React.ReactElement =
+    R.from MarkerClusterer (keyValueList CaseRules.LowerFirst props) markers
+
+let GoogleMapComponent: RCom = import "GoogleMapComponent" "./js/mapComponent.js"
+
+
+/// A wrapper around google.maps.Map
+let googleMap (props:Props.IMapProperties list) : React.ReactElement =
+
+    R.from GoogleMapComponent (keyValueList CaseRules.LowerFirst props) []
+
+
+//
+// export function getPosition (options) {
+//     return new Promise(function (resolve, reject) {
```

Comment:
what's the easiest way to write that function directly in fable?

Reply:
You can use [Promise.create](https://github.com/fable-compiler/fable-powerpack/blob/1d107dc8312b38812947df975b2172f1de0ef2e1/src/Promise.fs#L17) from Fable.PowerPack.